### PR TITLE
Added improvements for displaying service name under service icon

### LINKED ISSUE #2151: "Display service name under icon" setting needs some improvements
**Bug description:**
As can be seen in the attached screenshots below, both horizontal and vertical bars have issues.

**Steps to reproduce:**

1. Go to "Settings"
2. Enable "Display service name under the icon"
4. See layout misalignments

**Expected behavior**
- Service name not to be overlapped by notification badge
- No scrollbars for services on horizontal bar

**Screenshots**
Screenshot from 2021-10-27 09-35-10

Screenshot from 2021-10-27 09-34-45

**Environment:**

- Operating System: Ubuntu 21.10
- Ferdi Version: 5.6.3-nightly.44
- Server: Ferdi

### PR DESCRIPTION
**Pull request by AnjithPaul**
Co-authored-by: @johan-mathew  johankmathew@gmail.com
#### Pre-flight Checklist

1. Please remember that if you are logging a bug for some service that has _stopped working_ or is _working incorrectly_, please log the bug [here](https://github.com/getferdi/recipes/issues)
2. If you are requesting support for a **new service** in Ferdi, please log it [here](https://github.com/getferdi/recipes/pulls)
3. Please remember to read the [self-help documentation](https://github.com/getferdi/ferdi#troubleshooting-recipes-self-help) - in case it helps you unblock yourself for issues related to older versions of recipes that were installed on your machine. (These will get automatically upgraded when you upgrade to the newer versions of Ferdi, but to get new recipes between Ferdi releases, this documentation is quite useful.)
4. Please consider supporting Ferdi!
  👉 https://github.com/sponsors/getferdi
  👉 https://opencollective.com/getferdi/donate
5. Please ensure you've completed all of the following.

- [x] I have read the [Contributing Guidelines](https://github.com/getferdi/ferdi/blob/develop/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/getferdi/ferdi/blob/develop/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://github.com/getferdi/ferdi/issues) for a feature request that matches the one I want to file, without success.

#### Description of Change

1. Changed notification badge position in-order to not overlap with service icon label.
2. Changed horizontal sidebar height from hard coded to change according to the size specified in the settings.
3. Removed scrollbar from the service icon in horizontal sidebar.

#### Motivation and Context

Displaying service name under service icon needed some improvements to become more user friendly. Fixes #2151

#### Screenshots

Screenshot 2021-10-31 at 12 33 07 PM

#### Checklist

- [x] My pull request is properly named
- [x] The changes respect the code style of the project (`npm run prepare-code`)
- [x] `npm test` passes
- [x] I tested/previewed my changes locally

### CODE REVIEW

File: src/features/appearance/index.ts [hunk 1]
```diff
@@ -114,31 +114,99 @@ function generateAccentStyle(accentColorStr) {
...
   return vertical
     ? `
+    .sidebar {
+      height: ${width}px !important;
+      overflow: hidden !important;
+    }
     .tab-item {
       width: ${width - 2}px !important;
-      height: ${width - 5 + iconSize}px !important;
+      height: ${width - 5 + iconSize +tabItemSizeBias}px !important;
```
Comment (vraravam) [inline src/features/appearance/index.ts:174]: Please format correctly.

File: src/styles/vertical.scss [hunk 2]
```diff
@@ -51,7 +67,7 @@ $sidebar-width: 64px;
 }
 
 .app .app__content {
-  padding-top: $sidebar-width;
+  padding-top: $sidebar-width+22;
```
Comment (vraravam) [inline src/styles/vertical.scss:70]: Can this magic number be from a variable?

File: src/styles/vertical.scss [hunk 3]
```diff
@@ -91,4 +107,7 @@ $sidebar-width: 64px;
   .sidebar__button--workspaces.is-active {
     height: $sidebar-width - 22px;
   }
+  .tab-item .is-label-enabled{
+    height: $sidebar-width +30px;
```
Comment (vraravam) [inline src/styles/vertical.scss:111]: Can this magic number be from a variable?

**Review by vraravam — CHANGES_REQUESTED** [review]

File: src/features/appearance/index.ts (hunk 1 above)
Comment (AnjithPaul) [reply src/features/appearance/index.ts:174]: okay

File: src/styles/vertical.scss (hunk 2 above)
Comment (AnjithPaul) [reply src/styles/vertical.scss:70]: That number was already present in the code. We can extract it to a variable.

File: src/styles/vertical.scss (hunk 3 above)
Comment (AnjithPaul) [reply src/styles/vertical.scss:111]: That number was already present in the code. We can extract it to a variable.

File: src/styles/vertical.scss [hunk 4]
```diff
@@ -30,17 +31,32 @@ $sidebar-width: 64px;
     .tab-item {
       
       &.is-active {
-        border-left-width: 0px;
+        border-left-width: 0px !important;
         border-top-width: 4px;
         border-top-style: solid;
-        padding-left: 4px;
+        overflow: hidden;
+        height: $sidebar-width+4;
```
Comment (vraravam) [inline src/styles/vertical.scss:38]: formatting?

File: src/styles/vertical.scss [hunk 5]
```diff
@@ -30,17 +31,32 @@ $sidebar-width: 64px;
...
       }
 
       &:not(.is-active) {
         padding-top: 4px;
       }
+
+      &.is-label-enabled{
+        min-width: 70px;
+        max-width: 140px;
+        height: $sidebar-width +10;
```
Comment (vraravam) [inline src/styles/vertical.scss:48]: formatting

File: src/styles/vertical.scss (hunk 2 above)
Comment (vraravam) [reply src/styles/vertical.scss:70]: boy scout rule: keep it better than before.

File: src/features/appearance/index.ts [hunk 6]
```diff
@@ -114,31 +114,99 @@ function generateAccentStyle(accentColorStr) {
...
+      overflow: hidden !important;
     }
     .tab-item .tab-item__icon {
       width: ${width / 2 + iconSize}px !important;
     }
     .sidebar__button {
       font-size: ${width / 3}px !important;
     }
+    .app .app__content {
+      padding-top: ${width + 22 + sidebarSizeBias}px !important;
```
Comment (vraravam) [inline src/features/appearance/index.ts:185]: magic number

File: src/features/appearance/index.ts [hunk 7]
```diff
@@ -114,31 +114,99 @@ function generateAccentStyle(accentColorStr) {
...
+    }
+    .workspaces-drawer {
+      margin-top: -${width}px !important;
+    }
+    .darwin .sidebar {
+      height: ${22 + width + sidebarSizeBias}px !important;
+    }
+    .darwin .sidebar .sidebar__button--workspaces.is-active {
+      height: ${width - 22}px !important;
```
Comment (vraravam) [inline src/features/appearance/index.ts:194]: magic number

**Review by vraravam — CHANGES_REQUESTED** [review]

**Review by vraravam — APPROVED** [review]

**Pull request merged**